src/server/dflycmd.cc

```diff
@@ -700,14 +636,22 @@ shared_ptr<DflyCmd::ReplicaInfo> DflyCmd::GetReplicaInfo(uint32_t sync_id) {
 
 std::vector<ReplicaRoleInfo> DflyCmd::GetReplicasRoleInfo() const {
   std::vector<ReplicaRoleInfo> vec;
-  unique_lock lk(mu_);
+  lock_guard lk(mu_);
 
+  vec.reserve(replica_infos_.size());
   auto replication_lags = ReplicationLags();
 
   for (const auto& [id, info] : replica_infos_) {
+    LSN lag = replication_lags[id];
+
+    lock_guard repl_lk{info->mu};
+
```

Comment:
the problem here is that the full sync -> stable sync transition might take some time (we should actually measure it), so even if only one replica takes unusually longer, your stats will freeze for _all_ replicas... Given we do it in GetMetrics(), INFO ALL will freeze as well